### DISCUSSION TITLE
Inverting buttons that are normally closed

### DISCUSSION THREAD

**Discussion by MoSaye** [tttapa/Control-Surface#954] [General]
hello there, I have a few touch sensors that are normally closed so when they are touched they send a MIDI note off and when released they send a MIDI note on, how do I invert them for certain buttons because I already have some other mechanical switches that are working fine, I saw in the documentation something about invert but there are no examples so I don't know how to use it, can anyone help?

**Comment by tttapa** [comment]
See [`NoteButton::invert()`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d1/d9f/classNoteButton.html#a7fa1616cc61c19a5efcc863c950f7f30).

**Reply by MoSaye** [reply]
Thanks for the fast reply, and I apologize for my noobness, as I am an artist, not a developer however, I saw this before writing the note and I couldn't figure out how to use it since there was no example of how to implement it, can you just get me to the start of the way and i could go on from there, thanks again.

**Reply by tttapa** [reply]
If you start from the NoteButton example, you can simply add `button.invert();` to your `setup()` function.

**Reply by MoSaye** [reply]
thanks a lot for your help it worked even with multiplexer i just dropped the address of the switch and it worked, and i would like to thank you for the great work that you are doing here.

**Reply by rommil67** [reply]
How to do this for many (35) such buttons in Arduino Mega? code snippet (I use old examples):

NoteButton buttons[] = {

    
    // If you want to set the channel specifically, you can use:
    // for UNO , nano, mega
    {7, {note(C, 2), CHANNEL_3}},
    {2, {note(Db, 2), CHANNEL_3}},

**Reply by tttapa** [reply]
Simply loop over all buttons:

```cpp
for (auto &b : buttons)
  b.invert();
```

**Reply by rommil67** [reply]
Thanx.
